pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -190,15 +189,32 @@ func (a *AmbientIndexImpl) All() []*model.AddressInfo {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 	res := make([]*model.AddressInfo, 0, len(a.byUID)+len(a.serviceByNamespacedHostname))
+
 	for _, wl := range a.byUID {
+		netAddrs := networkAddressFromWorkload(wl)
+
+		// Workload without an (optional) address is possible and should still be included
+		if len(netAddrs) == 0 {
```

Comment:
Can this happen, @stevenctl 

Feel this should have been populated with network gateway address. Need to okk carefully

Reply:
There is a dedicated field for network gateway

```
  GatewayAddress network_gateway = 19;
```

Reply:
If this has 0 addresses, it must have a hostname according to the proto comments. 
I think if neither are provided, we must have a NetworkGateway (or waypoint?)

zTunnel should also be smart about ignoring it if it has no way to use it

Reply:
can we have a test coverage for this case

Reply:
@hzxuzhonghu Do you mean test coverage for a scenario wiwth 0 addresses and no hostname?
This seems a bit off the topic the issue this PR addresses. Can we have it in a seperate one?

Reply:
how about a uint test now

Reply:
@hzxuzhonghu If you refer to 0 addresses it is covered by: https://github.com/istio/istio/blob/e81026debe6456e0121ffad715e8b56de02ad8ab/pilot/pkg/serviceregistry/kube/controller/ambientindex_workloadentry_test.go#L264

Reply:
It covers 0 addresses and https://github.com/istio/istio/blob/e81026debe6456e0121ffad715e8b56de02ad8ab/pilot/pkg/serviceregistry/kube/controller/ambientindex_serviceentry_test.go#L29 covers all the rest (at least from my coverage highlighting).

I think I don't fully follow what logic coverage you would like to be added.

Reply:
Looing at the test, it calls `AmbientIndexImpl.All` only when s.lookup(key =""), however where is the call that key = ""

Reply:
NVM After a careful look, i see L269 and L273 both call with ket = ""

Reply:
I doubt the test still pass  even with the new logic

Reply:
The logic returns the WL even if there are no addresses and L269 / 273 make sure lookup returned them.
Why do you think it should fail?

Reply:
My fault， the check is have no diff with previous, just a skip for below processing